aws/resource_aws_ssm_document.go

```diff
@@ -171,17 +166,25 @@ func resourceAwsSsmDocument() *schema.Resource {
 			"target_type": {
 				Type:     schema.TypeString,
 				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^\/[\w\.\-\:\/]*$`), ""),
+					validation.StringLenBetween(1, 200),
+				),
 			},
 			"version_name": {
 				Type:     schema.TypeString,
 				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_\-.]{3,128}$`), ""),
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16628

```suggestion
					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_\-.]{3,128}$`), "must contain only alphanumeric, underscore, hyphen, or period characters"),
```